packages/um/package.py

```diff
@@ -152,24 +152,14 @@ class Um(Package):
             "fcm_name": "netcdf",
             "fcm_ld_flags": "-lnetcdff -lnetcdf"}}
 
-    # Source acquisition for AM3
-    resource(
-        name="jules_sources",
-        git="https://github.com/ACCESS-NRI/JULES.git",
-        branch="AM3-dev",
-        when="model=\"vn13p1-am\"",
-        destination="resources",
-        placement=dict(JULES="JULES")
-    )
-
-    resource(
-        name="um_sources",
-        git="https://github.com/ACCESS-NRI/UM.git",
-        branch="AM3-dev",
-        when="model=\"vn13p1-am\"",
-        destination="resources",
-        placement=dict(UM="UM")
-    )
+    # Optional JULES sources to be used in build (i.e. AM3)
+    jules_git_url = "git@github.com:ACCESS-NRI/JULES.git"
+    variant("jules_sources", default="NA", values=str, multi=False, description=f"Optional JULES sources branch/tag/commit from {jules_git_url}.")
+    
+    # Optional UM sources to be used in build (i.e. AM3)
+    um_git_url = "git@github.com:ACCESS-NRI/UM.git"
+    variant("um_sources", default="NA", values=str, multi=False, description=f"Optional UM sources branch/tag/commit from {um_git_url}.")
```

Comment:
These are `git` URLs not HTTPS. I think we need to use HTTPS and a token for the build-cd (and service user) to be able to checkout the private sources. I defer to @CodeGat however.